pkg/devfile/resources.go

```diff
@@ -0,0 +1,258 @@
+		if _, ok := deployAssociatedComponents[component.Name]; ok && component.Kubernetes != nil {
+			var src parser.YamlSrc
+
+			if component.Kubernetes.Inlined != "" {
+				src = parser.YamlSrc{
+					Data: []byte(component.Kubernetes.Inlined),
+				}
+			} else {
+				// If the Kubernetes YAML file does not have inline content, fall back to the dev preview way
+				// Once we have resolved the flow between frontend, backend and consume only YAML files,
+				// we can err out here instead of falling back
+
+				// Deployment
+				deploymentResource, err := GetDeployResource(name, devfileObj, common.DevfileOptions{})
+				if err != nil {
+					errMsg := fmt.Sprintf("Failed to get Deployment resource for the devfile: %v", err)
+					klog.Error(errMsg)
+					return appsv1.Deployment{}, corev1.Service{}, routev1.Route{}, err
+				}
+				appendedResources.Deployments = append(appendedResources.Deployments, deploymentResource)
+
+				dockerImagePort := devfileObj.Data.GetMetadata().Attributes.GetString("alpha.dockerimage-port", &err)
+				if err != nil {
+					errMsg := fmt.Sprintf("Failed to get the Dockerfile location from devfile metadata attribute 'alpha.build-dockerfile': %v", err)
+					klog.Error(errMsg)
+
+					return appsv1.Deployment{}, corev1.Service{}, routev1.Route{}, err
+				}
+
+				imageComponentFilter := common.DevfileOptions{
+					ComponentOptions: common.ComponentOptions{
+						ComponentType: devfilev1.ImageComponentType,
+					},
+				}
+
+				// Service
+				serviceResource, err := GetService(devfileObj, imageComponentFilter, dockerImagePort)
+				if err != nil {
+					errMsg := fmt.Sprintf("Failed to get service for the devfile: %v", err)
+					klog.Error(errMsg)
+					return appsv1.Deployment{}, corev1.Service{}, routev1.Route{}, err
+				}
+				appendedResources.Services = append(appendedResources.Services, serviceResource)
+
+				// Route
+				routeResource := GetRouteForDockerImage(name, dockerImagePort, "/", false, nil)
+				appendedResources.Routes = append(appendedResources.Routes, routeResource)
+
+				returnDeploymentOnly = false
+
+				// break here as we are no longer interested in parsing a YAML file
+				break
+			}
+
+			fs := afero.Afero{Fs: afero.NewOsFs()}
+			values, err := parser.ReadKubernetesYaml(src, fs)
+			if err != nil {
+				errMsg := fmt.Sprintf("Failed to read the Kubernetes yaml from devfile: %v", err)
+				klog.Error(errMsg)
+				return appsv1.Deployment{}, corev1.Service{}, routev1.Route{}, err
+			}
+
+			resources, err := parser.ParseKubernetesYaml(values)
+			if err != nil {
+				errMsg := fmt.Sprintf("Failed to parse the Kubernetes yaml data from devfile: %v", err)
+				klog.Error(errMsg)
+				return appsv1.Deployment{}, corev1.Service{}, routev1.Route{}, err
+			}
+
+			for _, endpoint := range component.Kubernetes.Endpoints {
+				if endpoint.Exposure != devfilev1.NoneEndpointExposure {
+					var isSecure bool
+					if endpoint.Secure != nil {
```

Comment:
you don't need to check for this. it's been done in parser already https://github.com/kim-tsao/library/blob/4c7c5dc73eee13b3db599a7e5fd2b266420cd390/pkg/devfile/parser/parse.go#L702

Reply:
I would still need to check if its nil because I am accessing it like *endpoint.Secure; I didnt check initially and it panicked in one of my tests with an empty `secure`

Reply:
in unit test, yes, you can assign a nil to `endpoint.Secure`. but in actual case, all boolean values will be set before returning `devfileObj`